src/automaton/mod.rs

```diff
@@ -98,22 +104,27 @@ pub trait Automaton {
 impl<'a, T: Automaton> Automaton for &'a T {
     type State = T::State;
 
+    #[inline]
```

Comment:
These are generic methods right, so we don't need them?

Reply:
I probably forgot to remove these... I'll fix it tomorrow...